File: 150_evaluate_reverse_polish_notation.js
/**
 * @param {string[]} tokens
 * @return {number}
 */
var evalRPN = function (tokens) {
    let myStack = []
    for (let token of tokens) {
        if (token === "+") {
            let a = myStack.pop()
            let b = myStack.pop()
            myStack.push(a + b)
        }
        else if (token === "-") {
            let a = myStack.pop()
            let b = myStack.pop()
            myStack.push(b - a)
        }
        else if (token === "*") {
            let a = myStack.pop()
            let b = myStack.pop()
            myStack.push(b * a)
        }
        else if (token === "/") {
            let a = myStack.pop()
            let b = myStack.pop()
            myStack.push(Math.trunc(b / a))
        } else {
            myStack.push(parseInt(token))
        }
    }
    return myStack.pop()
};

File: 150_evaluate_reverse_polish_notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        my_stack = []
        for r in tokens:
            if r not in {"+", "-", "/", "*"}:
                my_stack.append(int(r))
            else:
                a = my_stack.pop()
                b = my_stack.pop()
                if r == "+":
                    my_stack.append(a + b)
                if r == "-":
                    my_stack.append(b - a)
                if r == "*":
                    my_stack.append(b * a)
                if r == "/":
                    my_stack.append(int( b / a))
        return my_stack[0]